module/expressions/src/commonMain/kotlin/org/cru/godtools/expressions/Expression.kt

```diff
@@ -30,3 +33,11 @@ fun String.toExpressionOrNull() = when {
         }
     )
 }
+
+private fun ParserRuleContext.vars(): Collection<String>? = children?.flatMap {
+    when (it) {
+        is TerminalNode -> listOfNotNull(it.symbol?.takeIf { it.type == StateExpressionParser.Tokens.VAR.id }?.text)
+        is ParserRuleContext -> it.vars()
+        else -> null
```

Comment:
Could you explain this logic to me? Not sure I know what this is doing fully. Also, should there be a test for line 41?